integration-tests/suites/gperftools.go

```diff
@@ -0,0 +1,78 @@
+package suites
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"golang.org/x/sys/unix"
+)
+
+type GperftoolsTestSuite struct {
+	IntegrationTestSuiteBase
+}
+
+// IsArchSupported: at the moment only x86_64 builds have gperftools
+func (s *GperftoolsTestSuite) IsArchSupported() (bool, string) {
+	u := unix.Utsname{}
+	unix.Uname(&u)
+
+	// Exclude null bytes from comparison
+	arch := string(bytes.Trim(u.Machine[:], "\x00"))
+	return arch == "x86_64", arch
+}
+
+func (s *GperftoolsTestSuite) SetupSuite() {
+	if ok, _ := s.IsArchSupported(); !ok {
+		return
+	}
+
+	s.StartCollector(false, nil)
+}
+
+func (s *GperftoolsTestSuite) TearDownSuite() {
+	if ok, _ := s.IsArchSupported(); !ok {
+		return
+	}
+
+	s.StopCollector()
+}
+
+// Verify Gperftools API:
+// * turn heap profiling on
+// * turn heap profiling off
+// * fetch the profile
+//
+// NOTE: The test will only be performed on supported architectures, as defined
+// by IsArchSupported.
+func (s *GperftoolsTestSuite) TestFetchHeapProfile() {
+	heap_api_url := "http://localhost:8080/profile/heap"
+	data_type := "application/x-www-form-urlencoded"
+
+	if ok, arch := s.IsArchSupported(); !ok {
+		fmt.Printf("[WARNING]: Skip GperftoolsTestSuite on %s\n", arch)
```

Comment:
You can do `s.T().Skip("[WARNING]: skip GperftoolsTestSuite on ", arch)`

Reply:
Haven't found anything immediately about how to do this in setup/teardown.